client/src/process.js

```diff
@@ -0,0 +1,213 @@
+/* eslint-disable no-undef */
+const baseURL = 'http://localhost:10000';
+
+/**
+ * Function displays the overlay
+ */
+
+export function showOverlay() {
+    document.getElementById('overlay').style.display = 'block';
+}
+
+/**
+ * Function hides the overlay
+ */
+
+export function hideOverlay() {
+    document.getElementById('overlay').style.display = 'none';
+}
+
+/**
+ * Function displays the loading spinner and overlay
+ */
+
+export function showSpinner() {
+    showOverlay();
+    document.getElementById('loadingSpinner').style.display = 'block';
+}
+
+/**
+ * Function hides the loading spinner and overlay
+ */
+
+export function hideSpinner() {
+    hideOverlay();
+    document.getElementById('loadingSpinner').style.display = 'none';
+}
+
+/**
+ * Function is called when the user clicks on the "Update" button
+ * Updates the network graph with the user ID entered in the input box
+ */
+
+export function updateID() {
+    console.log('Update ID called');
+    showSpinner();
+
+    let userID = document.getElementById('idInput').value;
+
+    if (!/^\d+$/.test(userID)) {
+        alert('Please enter a valid number.');
+        console.log('Invalid user ID entered');
+        hideSpinner();
+        return;
+    }
+
+    console.log('User ID', userID);
+
+    let apiURL = `${baseURL}/api/v1/account/${userID}/initialize`;
+
+    // Need to write a common http request function with error handling
+    try {
+        fetch(apiURL)
+            .then((response) => response.json())
+            .then((json) => {
+                try {
+                    console.log(json);
+                    const [nodes, edges] = processData(json, userID);
+                    console.log('Nodes and edges', nodes, edges);
+                    drawFromResponse(nodes, edges);
+                    hideSpinner();
+                } catch (error) {
+                    console.log('Error in processing response', error);
+                    // Handle or rethrow the error as needed
+                }
+            });
+    } catch (error) {
+        alert('Error fetching data from API. Please try again later.');
+        console.log('Error', error);
+        hideSpinner();
+    }
+}
+
+/**
+ *
+ * @param {JSON} json Response json from the API
+ * @param {Number} mainID
+ * @returns {Array} [nodes,edges]
+ *
+ * Function takes in the JSON response from the API and the main user ID
+ * Returns an array of nodes and edges to be used in the network graph
+ *
+ *
+ */
+
+export function processData(json, mainID = '109252111498807689') {
+    nodes_json = json.accountInfoList;
+    edges_json = json.relations;
+
+    main_node = nodes_json.filter((node) => node.id == mainID)[0];
+
+    let nodes = [];
+    let nodesCreated = new Set(); // to keep track of nodes created using the ids of accounts
+    nodes.push({
+        id: main_node.id,
+        value: 20,
```

Comment:
Meaning of value is unclear here

Reply:
the value is related to the size of the node, but actually we can try to remove it (and see if it still work) for this PR since that is another feature

Reply:
Good comment, added comment explaining it's for size of node